crypto/pem/pem_lib.c

```diff
@@ -971,19 +971,25 @@ int PEM_read_bio_ex(BIO *bp, char **name_out, char **header,
     headerlen = BIO_get_mem_data(headerB, NULL);
     *header = pem_malloc(headerlen + 1, flags);
     *data = pem_malloc(len, flags);
-    if (*header == NULL || *data == NULL) {
-        pem_free(*header, flags, 0);
-        pem_free(*data, flags, 0);
-        goto end;
+    if (*header == NULL || *data == NULL)
+        goto out_free;
+    if (BIO_read(headerB, *header, headerlen) != headerlen) {
```

Comment:
Then just replace this condition with `headerlen != 0 && BIO_read(headerB, *header, headerlen) != headerlen`

Reply:
There is no point in calling the BIO_read for 0 bytes.